ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/toolbarview/MessengerToolbarFragmentTest.java

```diff
@@ -0,0 +1,570 @@
+        PowerMockito.when(messengerToolbarExpandedFragment, IS_ADDED_METHOD).thenReturn(true);
+        PowerMockito.when(messengerToolbarExpandedFragment, GET_VIEW_METHOD).thenReturn(view);
+
+        PowerMockito.suppress(PowerMockito.methods(Fragment.class, ON_CREATE_METHOD));
+        PowerMockito.suppress(PowerMockito.methods(Fragment.class, ON_VIEW_CREATED_METHOD));
+        PowerMockito.suppress(PowerMockito.methods(Fragment.class, ON_ACTIVITY_CREATED_METHOD));
+        PowerMockito.suppress(PowerMockito.methods(Fragment.class, ON_DESTROY_VIEW_METHOD));
+        PowerMockito.suppress(PowerMockito.methods(MessengerTemplateHelper.class, APPLY_BACKGROUND_THEME_METHOD));
+    }
+
+    @Test
+    public void onCreate() {
+        // Arrange & Act
+        messengerToolbarFragment.onCreate(bundle);
+    }
+
+    @Test
+    public void onCreateView() {
+        // Arrange & Act
+        View viewResult = messengerToolbarFragment.onCreateView(inflater, container, bundle);
+
+        // Assert
+        assertNotNull(viewResult);
+    }
+
+    @Test
+    public void onViewCreated() {
+        // Arrange
+        messengerToolbarFragment.onCreateView(inflater, container, bundle);
+
+        // Act
+        messengerToolbarFragment.onViewCreated(view, bundle);
+    }
+
+    @Test
+    public void onActivityCreated() {
+        // Arrange
+        messengerToolbarFragment.onCreate(bundle);
+
+        // Act
+        messengerToolbarFragment.onActivityCreated(bundle);
+    }
+
+    @Test
+    public void onDestroyView() {
+        // Arrange
+        messengerToolbarFragment.onCreate(bundle);
+
+        // Act
+        messengerToolbarFragment.onDestroyView();
+    }
+
+    @Test
+    public void configureDefaultView() {
+        // Arrange
+        messengerToolbarFragment.onCreate(bundle);
+
+        // Act
+        messengerToolbarFragment.configureDefaultView();
+    }
+
+    @Test
+    public void configureForLastActiveUsersViewWhenExpanded() {
+        // Arrange
+        prepareMessengerToolbarFragment();
+
+        Whitebox.setInternalState(messengerToolbarFragment, IS_EXPANDED_FIELD, true);
+
+        // Act
+        messengerToolbarFragment.configureForLastActiveUsersView(lastActiveAgentsData, true);
+    }
+
+    @Test
+    public void configureForLastActiveUsersViewWhenCollapsed() {
+        // Arrange
+        prepareMessengerToolbarFragment();
+
+        Whitebox.setInternalState(messengerToolbarFragment, IS_EXPANDED_FIELD, false);
+
+        // Act
+        messengerToolbarFragment.configureForLastActiveUsersView(lastActiveAgentsData, true);
+    }
+
+    @Test
+    public void configureForLastActiveUsersViewWhenViewNotReady() throws Exception {
+        // Arrange
+        prepareMessengerToolbarFragment();
+
+        PowerMockito.when(fragmentHostCallback, GET_ACTIVITY_METHOD).thenReturn(null);
+
+        // Act
+        messengerToolbarFragment.configureForLastActiveUsersView(lastActiveAgentsData, true);
```

Comment:
🚫  This test doesn't contain a mechanism to assert the expected results or exceptional cases. This kind of tests creates artificial code coverage. To consider a test as valid, it must be prone to fail and it must have either assertions or expected exceptions. Consider adding assertions or expect specific exceptions in order to follow testing best practices.  

**[More details](https://confluence.devfactory.com/x/AxqkEw)**
          
## The solution in this case
*Proof of concept*
```JAVA
//TODO: Adding result and state change asserts (and verify calls if appropriate).
// If there are no results consider adding more relevant assertions that focus on testing
// the side effects of this operation
assertNotNull(messengerToolbarFragment.get(property));

```